libvmi/driver/xen/xen_events_private.h

```diff
@@ -153,30 +154,10 @@ static const unsigned int event_response_conversion[] = {
     [VMI_EVENT_RESPONSE_NEXT_SLAT_ID] = VM_EVENT_FLAG_FAST_SINGLESTEP,
 };
 
-static inline status_t
-vmi_flags_sanity_check(vmi_mem_access_t page_access_flag)
-{
-    /*
-     * Setting a page write-only or write-execute in EPT triggers and EPT misconfiguration error
-     * which is unhandled by Xen (at least up to 4.3) and instantly crashes the domain on the first trigger.
-     *
-     * See Intel® 64 and IA-32 Architectures Software Developer’s Manual
-     * 28.2.3.1 EPT Misconfigurations
-     * AN EPT misconfiguration occurs if any of the following is identified while translating a guest-physical address:
-     * * The value of bits 2:0 of an EPT paging-structure entry is either 010b (write-only) or 110b (write/execute).
-     */
-    if (page_access_flag == VMI_MEMACCESS_R || page_access_flag == VMI_MEMACCESS_RX) {
-        errprint("%s error: can't set requested memory access, unsupported by EPT.\n", __FUNCTION__);
-        return VMI_FAILURE;
-    }
-
-    return VMI_SUCCESS;
-}
-
 static inline status_t
 convert_vmi_flags_to_xenmem(vmi_mem_access_t page_access_flag, xenmem_access_t *access)
 {
-    if ( VMI_FAILURE == vmi_flags_sanity_check(page_access_flag) )
+    if ( VMI_FAILURE == intel_mem_access_sanity_check(page_access_flag) )
```

Comment:
I guess at some point we also want to ensure this check is only performed when we are actually running on x86.